### PR TITLE
Fix minStay-maxStay and fix today's cell

### PR DESCRIPTION
**Pull request by okzea**
- min/max stay: get the biggest value from the range instead of first cell's
- fix issue when yesterday's cell is available and today's not
- fix issue on today's cell when the calendar has no rental
- remove unused code
- refactor currentDate

Demo from local:
image

Screen Recording 2021-09-08 at 03 32 57 11 PM

### CODE REVIEW

File: src/calendar.js [hunk 1]
```diff
@@ -58,13 +58,32 @@ export default class Calendar extends Emitter {
...
       this.opts.startOfWeek = this.opts.startOfWeek || this.locale.startOfWeek;
+
+      let { currentDate } = this.opts;
+      currentDate = currentDate.toLocaleString('en', {
+        year: 'numeric',
+        month: '2-digit',
+        day: '2-digit'
+      }).split('/');
+      const year  = parseInt(currentDate[2]);
+      const month = parseInt(currentDate[0]) - 1;
```
Comment (abel-n) [inline src/calendar.js:69]: I see this piece of code had just been added, so let's do what we talked about and use native Date functions:
```suggestion
      const year  = currentDate.getFullYear();
      const month = currentDate.getMonth();
```

File: src/config.js [hunk 2]
```diff
@@ -21,15 +21,13 @@ export default {
 
   startOfWeek: 0, // 0 Sun 1 Mon....6 Sat
   minStay: 1, // can select one night
-  maxStay: 0, // can select infinite nights
-  monthStart: currDate.getUTCMonth(), // start with current month by default M '0...12'
-  yearStart: currDate.getUTCFullYear(), // start with current year YYYY
+  maxStay: Infinity, // can select infinite nights
+  monthStart: null, // start with current month by default M '0...12'
+  yearStart: null, // start with current year YYYY
```
Comment (abel-n) [inline src/config.js:26]: I'd probably have `maxStay: null` as default value, then in the code you can do `Number.isFinite(opts.maxStay)` to determine if something's been passed (isFinite will return true for all numbers, including `0` but excluding `NaN`, `Infinity` and `-Infinity`).

Reason is it's very strange to pass `Infinity` as an init option, or, rather, as any parameter. :smile: Or maybe it's just me who hasn't seen enough of it.

File: src/calendar.js [hunk 3]
```diff
@@ -480,16 +496,30 @@ export default class Calendar extends Emitter {
...
+            return cell.dataset.minStay || minStay;
+          }));
+        }
+
+        if (!opts.allowLongerMaxStaySelection) {
+          // get the biggest maxStay value in the range
+          maxStay = Math.max(...range.map(function(cell) {
+            return cell.dataset.maxStay || maxStay;
+          }));
+        }
```
Comment (abel-n) [inline src/calendar.js:518]: AFAICT this means that there is the possibility to have number of minimum / maximum nights defined for each day, and not just globally? That would introduce some interesting problems.

**Review by abel-n — CHANGES_REQUESTED** [review]
Not many things to mention:

File: src/calendar.js (hunk 3 above)
Comment (okzea) [reply src/calendar.js:518]: @abel-n that's a good question, by doing this you could have a global minStay of 5 but one period from bookings data that would have a minStay of 10. But if you allow allowShorterMinStaySelection that would use the global minStay so I guess it's more flexible. ````

File: src/config.js (hunk 2 above)
Comment (okzea) [reply src/config.js:26]: @abel-n I get the point, IMO this is easier to read and keeps code shorter by removing unnecessary value checks  ` `

Comment (abel-n) [reply src/config.js]: I also see your point about the code having ~20 less characters, however, it's still very strange to pass `Infinity` to any function. It makes it a bit better that this is part of the default config, so ideally users would never actually pass `Infinity`. Notwithstanding, I'd have 20 more characters in code and have the cleanest possible "signature" of constructor / config. I'll let you decide which provides better ergonomics in the long run.

**Review by abel-n — APPROVED** [review]
:heavy_check_mark: I approve this message! :100:

**Pull request merged**